### PR TITLE
Merge GalleryPreviewScreen to GalleryDetailScreen

### PR DESCRIPTION
**Pull request by FooIbar**

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt [hunk 1]
```diff
@@ -0,0 +1,810 @@
...
+        navigate(lub.asDst())
+    }
+    fun onUploaderChipClick(galleryInfo: GalleryInfo) {
+        val uploader = galleryInfo.uploader
+        val disowned = uploader == "(Disowned)"
+        if (uploader.isNullOrEmpty() || disowned) {
+            return
+        }
+        val lub = ListUrlBuilder()
+        lub.mode = ListUrlBuilder.MODE_UPLOADER
```
Comment (revonateB0T) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt:206]: Inline property assign to ctor.

File: app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt [hunk 2]
```diff
@@ -0,0 +1,810 @@
...
+    if (showBottomSheet && galleryDetail != null) {
+        ModalBottomSheet(
+            onDismissRequest = { showBottomSheet = false },
+            contentWindowInsets = { EmptyWindowInsets },
+        ) {
+            GalleryInfoBottomSheet(galleryDetail)
+        }
+    }
+
+    fun onGalleryInfoCardClick() {
```
Comment (revonateB0T) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt:222]: Better convert to use `dialogState`

File: app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt [hunk 3]
```diff
@@ -0,0 +1,810 @@
...
+
+    val filterAdded = stringResource(R.string.filter_added)
+    fun showFilterUploaderDialog(galleryInfo: GalleryInfo) {
+        val uploader = galleryInfo.uploader
+        val disowned = uploader == "(Disowned)"
+        if (uploader.isNullOrEmpty() || disowned) {
+            return
+        }
+        launchIO {
+            awaitPermissionOrCancel {
```
Comment (revonateB0T) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt:234]: I'd suggest we rename to `awaitConfirmationOrCancel`

File: app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt [hunk 4]
```diff
@@ -0,0 +1,810 @@
...
+                Text(text = stringResource(R.string.filter_the_uploader, uploader))
+            }
+            Filter(FilterMode.UPLOADER, uploader).remember()
+            showSnackbar(filterAdded)
+        }
+    }
+    val onDownloadButtonClick = rememberLambda(galleryInfo) {
```
Comment (revonateB0T) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt:241]: We shuld investigate to remove this `rememberLambda` usage and definition, IIRC it is introduced with strongskipping mode, better to use @Dontmemorize

File: app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt [hunk 5]
```diff
@@ -0,0 +1,810 @@
...
+        val addFilter = stringResource(R.string.add_filter)
+        val filterAdded = stringResource(R.string.filter_added)
+        val upTag = stringResource(R.string.tag_vote_up)
+        val downTag = stringResource(R.string.tag_vote_down)
+        GalleryTags(
+            tagGroups = tags,
+            onTagClick = {
+                val lub = ListUrlBuilder()
+                lub.mode = ListUrlBuilder.MODE_TAG
+                lub.keyword = it
```
Comment (revonateB0T) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt:687]: Ditto.

File: app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt (hunk 3 above)
Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt:228]: That will affect too many files, gonna do that in another PR.

**Review by revonateB0T — APPROVED** [review]

**Pull request merged**